test/test_networking.py

```diff
@@ -1380,6 +1420,23 @@ def test_compat_opt_prefer_urllib(self):
             assert len(director.preferences) == 1
             assert director.preferences.pop()(UrllibRH, None)
 
+    def test_close(self, monkeypatch):
+        with FakeYDL() as ydl:
+            director = ydl._request_director
+            called = False
+            original_close = director.close
+
+            def mock_close(*args, **kwargs):
+                nonlocal called
+                called = True
+                return original_close(*args, **kwargs)
+
+            monkeypatch.setattr(director, 'close', mock_close)
+
+            ydl.close()
+
+            assert called is True
```

Comment:
I'm not sure if we would maybe want this? It asserts the cleanup occurs automatically when using `with` (`__exit__`). I would consider `.close()` convenience.
```suggestion
        assert called
```

Reply:
Removed tests for ytdl for now. We can deal with this in another PR imo